### ISSUE TITLE
Reggresion error: Repositories are processed on just one thread

### ISSUE THREAD

**Issue by ajnavarro** [src-d/engine#253]
Since commit https://github.com/src-d/engine/commit/c3c733831b2e6e138980fa925126edea42f5b7fc repositories are iterated using only one thread per worker. Can be the [group by](https://github.com/src-d/engine/commit/c3c733831b2e6e138980fa925126edea42f5b7fc#diff-cd0de52b340e0215d1274bb220523728R82) the problem?

**Comment by ajnavarro** [comment]
Definitely the problem is on the `groupBy`. For testing, I dropped all the logic inside the `groupBy`:

```scala
private def generateRDD(sc: SparkContext, path: String): RDD[RepositorySource] = {
   val test:RDD[RepositorySource] = sc.binaryFiles(s"$path/*")
      .map(_._2)
      .map(pds => {
        // returns a tuple of the root directory where it is contained, with a maximum depth
        // of 1 under the given path, the file name, and the portable data stream
        val path = pds.getPath
        val idx = path.indexOf('/', path.length + 1)
        if (idx < 0) {
          val p = new Path(path)
          (p.getParent.toString, p.getName, pds)
        } else {
          val (parent, file) = path.splitAt(idx)
          (parent, file, pds)
        }
      }).map {
     case (s1,s2,pds) =>
       SivaRepository(pds)
   }
    
    test
  }
```
and now it is working using all the threads.

**Comment by ajnavarro** [comment]
More relevant info to fix the issue: https://umbertogriffo.gitbooks.io/apache-spark-best-practices-and-tuning/content/avoid_groupbykey_when_performing_an_associative_re/avoid-groupbykey-when-performing-a-group-of-multiple-items-by-key.html

**Comment by ajnavarro** [comment]
fixed on v0.3.1
